packages/manager/modules/bm-server-components/src/ipmi/controller.js

```diff
@@ -0,0 +1,690 @@
+import isFunction from 'lodash/isFunction';
+import includes from 'lodash/includes';
+import set from 'lodash/set';
+
+import Ipmi from './ipmi.class';
+import Kvm from './kvm.class';
+
+import { getIpmiGuideUrl, ALERT_SECTION } from './constants';
+
+export default class BmServerIpmiController {
+  /* @ngInject */
+  constructor(
+    $sce,
+    $translate,
+    $window,
+    coreConfig,
+    IpmiService,
+    Polling,
+    Alerter,
+    $scope,
+    $q,
+  ) {
+    this.$sce = $sce;
+    this.$translate = $translate;
+    this.$window = $window;
+    this.coreConfig = coreConfig;
+    this.IpmiService = IpmiService;
+    this.Polling = Polling;
+    this.Alerter = Alerter;
+    this.$scope = $scope;
+    this.$q = $q;
+  }
+
+  $onInit() {
+    this.ttl = '5';
+    this.alertSection = ALERT_SECTION;
+    this.serviceName = this.server.name;
+    this.ipmi = null;
+    this.kvm = null;
+    this.canOrderKvm = false;
+
+    this.loader = {
+      loading: false,
+      error: false,
+      httpLoading: false,
+      httpError: false,
+      httpDone: false,
+      passwordLoading: false,
+      passwordError: false,
+      passwordDone: false,
+      pingLoading: false,
+      pingError: false,
+      pingDone: false,
+
+      buttonStart: false,
+      navigationLoading: false,
+      navigationReady: null,
+      javaLoading: false,
+      javaReady: false,
+      sshLoading: false,
+      kvmhtmlLoading: false,
+      solSshKeyLoading: false,
```

Comment:
are those infos really needed ? i'm not sure it's used

Reply:
yes all these are used.